### PR TITLE
[Schema] New help repo connection schema

### PR DESCRIPTION
**Pull request by Yashsharma1911**
**Description**

This PR fixes https://github.com/meshery/meshery/issues/8714

Added new RJSF schema for helm repo, with (name, description, a single URL)

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: schemas/configuration/helmRepoConnection.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        "title": "Name",
+        "minLength": 1,
+        "description": "Name of the new Helm repo connection",
+        "x-rjsf-grid-area" : "12"
+      },
+      "description": {
+        "type": "string",
+        "title": "Description",
+        "format": "textarea",
+        "description": "Description of this Helm repo connection.",
```
Comment (leecalcote) [inline schemas/configuration/helmRepoConnection.json:16]:
```suggestion
        "description": "An explanation as to the purpose and/or use of this Helm repository.",
```

File: schemas/configuration/helmRepoConnection.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "type": "object",
+    "properties": {
+      "name": {
+        "type": "string",
+        "title": "Name",
+        "minLength": 1,
+        "description": "Name of the new Helm repo connection",
```
Comment (leecalcote) [inline schemas/configuration/helmRepoConnection.json:9]:
```suggestion
        "description": "A short, memorable name of the Helm repository connection.",
```

File: schemas/configuration/helmRepoConnection.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        "title": "Description",
+        "format": "textarea",
+        "description": "Description of this Helm repo connection.",
+        "x-rjsf-grid-area" : "12"
+      },
+      "url": {
+        "type": "string",
+        "title": "URL",
+        "format": "uri",
+        "description": "The URL of the Helm repository where the charts are hosted.",
```
Comment (leecalcote) [inline schemas/configuration/helmRepoConnection.json:23]:
```suggestion
        "description": "The URL (must be http/s) of the Helm repository in which one or more Helm charts are hosted.",
```

File: schemas/configuration/helmRepoConnection.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        "x-rjsf-grid-area" : "12"
```
Comment (leecalcote) [inline schemas/configuration/helmRepoConnection.json:24]: Please include standard field validation for a URL.

File: schemas/configuration/helmRepoConnection.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      }
+    },
+    "required": [
```
Comment (leecalcote) [inline schemas/configuration/helmRepoConnection.json:27]: Thank you for making these required fields.

**Review by leecalcote — CHANGES_REQUESTED** [review]

**Comment by leecalcote** [comment]
Accepting suggested changes...

File: schemas/schemaProvider.go [hunk 6]
```diff
@@ -10,6 +10,7 @@ func getSchemaMap() map[string]string {
 		"filter":      "configuration/filterImport.json",
 		"design":      "configuration/designImport.json",
 		"publish":     "configuration/publishCatalogItem.json",
+		"helmRepo":    "configuration/helmRepoConnection.json",
```
Comment (leecalcote) [inline schemas/schemaProvider.go:13]: @Yashsharma1911, why is there no adjoining UISchema?

Comment (Yashsharma1911) [reply schemas/schemaProvider.go]: For this helmConnection schema there was not much use of it, but as we are maintaining files separately and to maintain consistency I added the ui file

**Comment by Yashsharma1911** [comment]
> Accepting suggested changes...

Thankyou for making changes direclty in PR

**Comment by Yashsharma1911** [comment]
This is how directory structure will look
```
🗂 Schemas
  |___configuration
  |___connections
         |___[ connectionName ]
                |___rjsfSchema.json
                |___uiRjsfSchema.json
```

**Pull request merged**